### DISCUSSION TITLE
Filament GlTF Loader aar Size

### DISCUSSION THREAD

**Discussion by Mehequanna** [google/filament#6687] [Q&A]
In my project, I was moving from `Sceneform` to `Sceneform Maintained`. This moved my Filament version from `1.9.8` to `1.21.1`. 

After checking the APK size, I noticed it was noticeably higher. The biggest change was `lib/libgltfio-jin.so`. It's about 5 mb larger. I've tracked this down to `Filament GlTF Loader` that went from `2.8mb` to `8mb` between Filament versions `1.9.8` to `1.21.1`. It's a large enough increase that there is discussion about how to make the APK smaller.

Is there a way to lower the size of `libgltfio-jin.so` while using Filament with `Sceneform Maintained`? Is there a lite version that would work with AR?

**Comment by romainguy** [comment]
This size increase unfortunately came from supporting more glTF features. There is a `lite` version of `gltfio` though: https://search.maven.org/artifact/com.google.android.filament/gltfio-android-lite/1.23.1/jar?eh=

It supports fewer glTF extensions, but the JNI is "only" 3 MiB.

**Reply by Mehequanna** [reply]
@romainguy thanks for the info. I'll give that a try. I just searched for some documentation on what the `lite` version supports and didn't find anything. Is there documentation on what the `lite` version  does or doesn't support in comparison to the full `gltfio`?

**Reply by romainguy** [reply]
I would have to dig through our archives. We don't provide `gltfio-lite` anymore unfortunately, as it became increasingly difficult to support the ever expanding glTF spec this way :/

**Reply by Mehequanna** [reply]
Thanks for the info! That makes sense. I can dig through some old commits. But it's likely we'd choose not to add something not in active development.

**Reply by romainguy** [reply]
Note that gltfio-lite has the same implementation (mostly, the difference was in the shaders it implements) and APIs as full gltfio.
